### PR TITLE
[VAULT-1441] Fix race that allowed remounting on path used by another mount

### PR DESCRIPTION
**Pull request by HridoyRoy**

### CODE REVIEW

**Review by swayne275 — APPROVED** [review]

**Comment by swayne275** [comment]
tests run on yours but not mine! jealous

**Comment by ncabatoff** [comment]
Please add a changelog entry.

**Comment by HridoyRoy** [comment]
> Please add a changelog entry.

Done!

File: changelog/11453.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+core: Fix for concurrent remounting of mount table entries.
```
Comment (ncabatoff) [inline changelog/11453.txt:2]:
```suggestion
core: Fix race that allowed remounting on path used by another mount
```

**Review by ncabatoff — APPROVED** [review]
Code looks good, added suggestion for more meaningful CL

**Pull request merged**